### DISCUSSION TITLE
how to use sortby with (translated array) with pagination

### DISCUSSION THREAD

**Discussion by YousufRizvi** [ppetzold/nestjs-paginate#850] [Q&A]
i have this data and want to apply sorting on name which is saved like this in db {"myVehicle en","myVehicle fr"}

`{
  "id": "string",
  "name": {
    "en": "string",
    "fr": "string"
  },
  "licenseNumber": "string",
  "manufacturer": "string",
  "modelYear": 0,
  "type": "string",
  "capacity": 0,
  "color": "string",
  "status": "string",
  "createdAt": "2024-01-09T12:38:11.325Z"
}`

**Comment by ppetzold** [comment]
what exactly is your question/problem? 😅

**Reply by YousufRizvi** [reply]
actually, this is my paginate config and i have a column name in which data is saved in two languages like english and french
I want to sort it by language like name[1] , is there anything i can do to achieve this?

from API its coming like this
```javascript
"name": { "en": "string", "fr": "string" }
``` 
thanks for your help

```typescript
import { FilterOperator, PaginateConfig } from 'nestjs-paginate'
import { Vehicle } from '../entity/vehicle.entity'

export const VEHICLE_PAGINATION_CONFIG: PaginateConfig<Vehicle> = {
  sortableColumns: ['name', 'licenseNumber'],
  nullSort: 'last',
  defaultSortBy: [['createdAt', 'DESC']],
  searchableColumns: ['name', 'licenseNumber'],
  filterableColumns: {
    type: [FilterOperator.IN],
    status: [FilterOperator.IN],
    createdAt: [FilterOperator.BTW],
  },
}

```

**Comment by ppetzold** [comment]
is this a JSON column, or embedded entity?

**Reply by YousufRizvi** [reply]
yes this is a column with type VARCHAR [] in postgres db

![Screenshot 2024-01-09 220657](https://github.com/ppetzold/nestjs-paginate/assets/58874989/185d1e13-ec34-4855-99fe-0be89554734f)

looks like this when fetched from db 

```javascript
Vehicle {
  id: 'd0f731e6-7d68-43bb-a9ab-03387c127576',
  name: [ 'myVehicle en', 'myVehicle fr' ],
  licenseNumber: null,
  manufacturer: null,
  modelYear: null,
  type: 'bus',
  capacity: null,
  color: null,
  status: 'active',
  createdAt: 2024-01-09T10:03:55.644Z,
  updatedAt: 2024-01-09T10:03:55.644Z,
  deletedAt: null
}
```

**Reply by AnnukirtiAxle** [reply]
I wanted to apply search on the array field but it's not working.

**Comment by ppetzold** [comment]
You can use `$contains` operator like `?filter.name=$contains:foobar` to filter on array columns

**Reply by YousufRizvi** [reply]
need to do something like this to sort by array indices, But I think it's not supported
```typescript

``` export const VEHICLE_PAGINATION_CONFIG: PaginateConfig<Vehicle> = {
  sortableColumns: ['name[0]', 'name[1]', 'licenseNumber'],
  // ... other properties
};

**Reply by ppetzold** [reply]
This is not supported. You are welcome to submit a feature PR.

Would check array length tho to avoid 500er:

```sql
SELECT *
FROM your_table
ORDER BY CASE WHEN array_length(your_array_column, 1) >= 2 THEN your_array_column[2] ELSE NULL END;
```

**Reply by YousufRizvi** [reply]
thanks , will submit soon
